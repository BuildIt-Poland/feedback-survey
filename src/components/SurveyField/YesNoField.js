import React from 'react';
import PropTypes from 'prop-types';

import { formField, radioOption } from '../../types';
import CustomRadioButtonGroup from '../CustomRadioButtonGroup';
import StyledFormField from './StyledFormField';

const YesNoField = ({ field, options }) => (
  <StyledFormField
    name={field.id}
    label={field.content}
    required={field.required}
    component={CustomRadioButtonGroup}
    options={options}
    width="medium"
    withIcon={false}
  />
);

YesNoField.propTypes = {
  field: formField.isRequired,
  options: PropTypes.arrayOf(radioOption)
};

export default YesNoField;
