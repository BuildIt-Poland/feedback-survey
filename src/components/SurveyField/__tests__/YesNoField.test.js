import React from 'react';
import { create } from 'react-test-renderer';

import YesNoField from '../YesNoField';

describe('COMPONENT - SurveyField YesNoField', () => {
  it('renders correctly', () => {
    const field = {
      id: '123',
      content: 'Question content',
      type: 'yesNo',
      required: false,
      options: ['Bad', 'Average']
    };
    const component = create(<YesNoField field={field} />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
