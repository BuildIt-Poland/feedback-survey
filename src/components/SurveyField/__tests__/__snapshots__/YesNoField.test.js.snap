// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - SurveyField YesNoField renders correctly 1`] = `
<div
  className="sc-ifAKCX ittBkw FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
>
  <label
    className="StyledText-sc-1sadyjn-0 bMXxti"
  >
    Question content
  </label>
  <div
    className="StyledBox-sc-13pk1d4-0 gZSidW"
  >
    <div
      className="sc-bxivhb iPHzoF"
    >
      <input
        className="sc-bdVaJa kRURFE"
        id="test123"
        name="fieldName"
        type="radio"
        value="test"
      />
      <label
        className="sc-htpNat kHxLtm"
        htmlFor="test123"
      >
        label-test
      </label>
    </div>
  </div>
</div>
`;
